tests/unit/analytics/test_analyze_results.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import logging
+from pyrit.analytics.analyze_results import analyze_results
+from pyrit.models.attack_result import AttackResult, AttackOutcome
+
+
+def make_attack(outcome: AttackOutcome) -> AttackResult:
+    return AttackResult(
+        conversation_id="test-id",
+        objective="test",
+        attack_identifier={"name": "mock"},
+        outcome=outcome,
+    )
+
+
+def test_empty_list_returns_none():
+    result = analyze_results([])
+    assert result["Attack success rate"] is None
+    assert result["Total decided"] == 0
+    assert result["Successes"] == 0
+    assert result["Failures"] == 0
+    assert result["Undetermined"] == 0
+
+
+def test_all_successes():
+    attacks = [make_attack(AttackOutcome.SUCCESS) for _ in range(5)]
```

Comment:
```suggestion
@pytest.mark.parametrize("n", [1, 2, 3, 10, 100])
def test_all_successes(n):
    attacks = [make_attack(AttackOutcome.SUCCESS) for _ in range(n)]
```
This creates a bunch of test cases with various numbers just to cover more. Same can be done with the next two test cases